### PR TITLE
WEB: Added dialog to select view when loading app-view

### LINKED ISSUE #28: Web Select AppView Dialog

### PR DESCRIPTION
**Pull request by Aagaard00**
Added dialog to select view when loading app-view
Added more info to BaseLoadMetaDataList

### CODE REVIEW

File: src/UICrafter.Client/Components/AppViewProperties.razor [hunk 1]
```diff
@@ -163,19 +165,23 @@
...
         try
         {
-            var test = await AppViewRepository.GetAppViews();
+            var dialog = DialogService.Show<AppViewSelectionDialog>();
+            var result = await dialog.Result;
 
-            var id = test.FirstOrDefault()?.Id ?? 0;
+            if (result is not null && !result.Canceled)
+            {
+                var id = (long)(result.Data ?? 0);
```
Comment (Aagaard00) [inline src/UICrafter.Client/Components/AppViewProperties.razor]: Dumbass, add error handling

File: src/UICrafter.Client/Components/LoadMetaDataWeb.razor [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+@code {
+    [Parameter]
+    public EventCallback<long> ClickFunction { get; set; }
+
+    protected override void HandleClick(AppView view)
+    {
+        if (ClickFunction.HasDelegate)
+        {
+            ClickFunction.InvokeAsync(view.Id);
```
Comment (Nephim) [inline src/UICrafter.Client/Components/LoadMetaDataWeb.razor]: Use the overload I made TryInvokeAsync HandleClick(AppView view) => ClickFunction.TryInvokeAsync (view.Id);

**Review by Gormiee — APPROVED** [review]
LGTM

**Review by Nephim — APPROVED** [review]
Rest of code looks fine.

**Pull request merged**